prow/repoowners/repoowners.go

```diff
@@ -215,36 +221,58 @@ func (c *Client) LoadRepoOwners(org, repo, base string) (RepoOwner, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	entry, ok := c.cache[fullName]
-	if !ok || entry.sha != sha || entry.owners == nil || entry.owners.enableMDYAML != mdYaml {
+	if !ok || entry.sha != sha || !entry.isValidOwners(mdYaml) {
 		gitRepo, err := c.git.Clone(cloneRef)
 		if err != nil {
 			return nil, fmt.Errorf("failed to clone %s: %v", cloneRef, err)
 		}
 		defer gitRepo.Clean()
-		if err := gitRepo.Checkout(base); err != nil {
-			return nil, err
-		}
 
-		if entry.aliases == nil || entry.sha != sha {
-			// aliases must be loaded
-			entry.aliases = loadAliasesFrom(gitRepo.Dir, log)
+		reusable := entry.isValid(mdYaml)
```

Comment:
so this is not really "valid" but a combination of 

1. we successfully loaded the owners last time (not missing aliases, etc)
2. prow config has not changed (mdyaml)

Would love `func (entry) fullyLoaded() bool` and `func (entry) matchesMDYAML(mdYAML bool) bool` to make this explicit

Reply:
fixed